README
GTKTerm 0.2.0
-------------

Keys for GTKTerm2 (v0.2.0)
--------------------------
CTRL + Page Down or KEYPAD RIGHT    next Terminal on the right hand
CTRL + Page Up or KEYPAD LEFT       next Terminal on the left hand
CTRL + T                            new Terminal

config ~/.gtkterm2.rc
---------------------

You can add different section. Each seaction needs a different match.
If you config your gtkterm2rc as follow and you su to root, all normal
letters are green.


##
# First section
##
[section]
match=root@
worldClass=-A-Za-z0-9/_:.,?+%=
scrollOnKeyStroke=1
scrollOnOutput=0
beep=1
blink=1
red=0x0000 0xcccc 0x0000 0xaaaa 0x0000 0xaaaa 0x6666 0xaaaa 0x0000 0x0000 0x5555 0xffff 0x5555 0xffff 0x5555 0xffff 0x5555 0xffff
grn=0x0000 0xcccc 0x0000 0x0000 0xaaaa 0x5555 0x6666 0x0000 0xaaaa 0xcccc 0x5555 0x5555 0xffff 0xffff 0x5555 0x5555 0xffff 0xffff
blu=0x0000 0xcccc 0x0000 0x0000 0x0000 0x0000 0xffff 0xaaaa 0xaaaa 0x0000 0x5555 0x5555 0x5555 0x5555 0xffff 0xffff 0xffff 0xffff




ChangeLog
2016-01-21  gettextize  <bug-gnu-gettext@gnu.org>

	* m4/gettext.m4: Upgrade to gettext-0.19.6.
	* m4/iconv.m4: Upgrade to gettext-0.19.6.
	* m4/lib-ld.m4: New file, from gettext-0.19.6.
	* m4/lib-link.m4: New file, from gettext-0.19.6.
	* m4/lib-prefix.m4: New file, from gettext-0.19.6.
	* m4/nls.m4: New file, from gettext-0.19.6.
	* m4/po.m4: New file, from gettext-0.19.6.
	* m4/progtest.m4: Upgrade to gettext-0.19.6.
	* Makefile.am (SUBDIRS): Remove intl.
	(ACLOCAL_AMFLAGS): New variable.
	(EXTRA_DIST): Add config.rpath.
	* configure.in (AM_GNU_GETTEXT_VERSION): Bump to 0.19.6.

2004-07-30 21:54  ofeige

	* po/de.gmo, po/de.po, src/interface.c, src/main.c, src/prefs.c,
	  src/prefs.h: Update opacity

2004-07-16 09:32  ofeige

	* configure.in: Update name from GTKTerm2 to gtkterm2

2004-07-13 23:37  ofeige

	* src/callbacks.c: Fixed compile warnings

2004-07-13 23:36  ofeige

	* src/main.c: add param -X -Y

2004-07-13 23:36  ofeige

	* src/interface.c: Fixed:  [ 977527 ] focus lost when
	  showTabsByOneTerminal=1  compile warnings

2004-06-08 14:53  ofeige

	* GTKTerm.glade, GTKTerm.gladep, GTKTerm.prj, Makefile.am,
	  Makefile.in, aclocal.m4, autogen.sh, config.h.in, configure,
	  configure.in, ltmain.sh, setup-gettext, po/de.gmo, po/de.po,
	  src/Makefile.am, src/Makefile.in: Update Project to Anjuta 1.2.2
	  Fix gettext 1.12.1 Bug

2004-06-08 14:52  ofeige

	* src/: callbacks.c, callbacks.h, interface.c, main.c, prefs.c,
	  prefs.h: Add stealth modus Add transparent Add argv Params for
	  Window positioning

2004-06-08 14:33  ofeige

	* GTKTerm.pws: no longer required

2004-02-20 00:04  ofeige

	* GTKTerm.pws, po/de.po, src/interface.c: Fix Version Number in

src/prefs.c
#ifdef HAVE_CONFIG_H
#  include <config.h>
#endif

#include <errno.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include <glib.h>
#include <gtk/gtk.h>
#include <gdk/gdk.h>
#include <gdk/gdkkeysyms.h>
#include <gtk/gtkprivate.h>

#include <vte/vte.h>

#include "prefs.h"

gtkTermPref* gtkTermPref_init(void)
{
	gtkTermPref* pref;

	pref = malloc(sizeof *pref);

	pref->notebook = NULL;
	pref->nbpage = 0;
	pref->notebookPanel = 2;
	pref->terminalScrollbar = 2;
	pref->maxScrollbackBuffer = 999;
	pref->quitPRGonExitLastTerminal = TRUE;
	pref->showTabsByOneTerminal = FALSE;
	pref->scrollOnKeyStroke = TRUE;
	pref->scrollOnOutput = FALSE;
	pref->blink = TRUE;
	pref->beep = TRUE;
	pref->termX = 80;
	pref->termY = 25;
	pref->winPosX = 0;
	pref->winPosY = 0;
	pref->opacity = 0;
	pref->transparent = FALSE;
	pref->stealth = FALSE;
	pref->login_shell = FALSE;
	pref->fullscreen = FALSE;

	pref->fore[0].red =   0xcccc;
	pref->fore[0].green = 0xcccc;
	pref->fore[0].blue =  0xcccc;

	pref->back[0].red =   0x0000;
	pref->back[0].green = 0x0000;
	pref->back[0].blue =  0x0000;

	pref->colors[0].red   = 0x0000;
	pref->colors[0].green = 0x0000;
	pref->colors[0].blue  = 0x0000;

	pref->colors[1].red   = 0xaaaa;
	pref->colors[1].green = 0x0000;
	pref->colors[1].blue  = 0x0000;

	pref->colors[2].red   = 0x0000;
	pref->colors[2].green = 0xaaaa;
	pref->colors[2].blue  = 0x0000;

	pref->colors[3].red   = 0xaaaa;
	pref->colors[3].green = 0x5555;
	pref->colors[3].blue  = 0x0000;

	pref->colors[4].red   = 0x6666;
	pref->colors[4].green = 0x6666;
	pref->colors[4].blue  = 0xffff;

	pref->colors[5].red   = 0xaaaa;
	pref->colors[5].green = 0x0000;
	pref->colors[5].blue  = 0xaaaa;

	pref->colors[6].red   = 0x0000;
	pref->colors[6].green = 0xaaaa;
	pref->colors[6].blue  = 0xaaaa;

	pref->colors[7].red   = 0xcccc;
	pref->colors[7].green = 0xcccc;
	pref->colors[7].blue  = 0xcccc;

	pref->colors[8].red   = 0x5555;
	pref->colors[8].green = 0x5555;
	pref->colors[8].blue  = 0x5555;

	pref->colors[9].red   = 0xffff;
	pref->colors[9].green = 0x5555;
	pref->colors[9].blue  = 0x5555;

	pref->colors[10].red   = 0x5555;
	pref->colors[10].green = 0xffff;
	pref->colors[10].blue  = 0x5555;

	pref->colors[11].red   = 0xffff;
	pref->colors[11].green = 0xffff;
	pref->colors[11].blue  = 0x5555;

	pref->colors[12].red   = 0x5555;
	pref->colors[12].green = 0x5555;
	pref->colors[12].blue  = 0xffff;

	pref->colors[13].red   = 0xffff;
	pref->colors[13].green = 0x5555;
	pref->colors[13].blue  = 0xffff;

	pref->colors[14].red   = 0x5555;
	pref->colors[14].green = 0xffff;
	pref->colors[14].blue  = 0xffff;

	pref->colors[15].red   = 0xffff;
	pref->colors[15].green = 0xffff;
	pref->colors[15].blue  = 0xffff;

	strncpy(pref->terminalFont, "Terminal 8", MAX_LINE_LENGTH - 1);
	strncpy(pref->worldClass, "-A-Za-z0-9/_:.,?+%=", MAX_LINE_LENGTH - 1);

	pref->mprefSize = 0;
	pref->mpref = NULL;

	gtkTermMPref_init(pref);

	return pref;
}

gtkTermMPref* gtkTermMPref_init(gtkTermPref* pref)
{
	pref->mprefSize++;
	pref->mpref = realloc(pref->mpref, sizeof *pref->mpref * (pref->mprefSize + 1));
	pref->mpref[pref->mprefSize] = malloc(sizeof **pref->mpref);

	pref->mpref[pref->mprefSize]->match[0] = '\0';
	pref->mpref[pref->mprefSize]->scrollOnKeyStroke = TRUE;
	pref->mpref[pref->mprefSize]->scrollOnOutput = FALSE;
	pref->mpref[pref->mprefSize]->beep = TRUE;
	pref->mpref[pref->mprefSize]->blink = TRUE;
	strncpy(pref->mpref[pref->mprefSize]->worldClass, "-A-Za-z0-9/_:.,?+%=", MAX_LINE_LENGTH - 1);

	pref->mpref[pref->mprefSize]->fore[0].red =   0xcccc;
	pref->mpref[pref->mprefSize]->fore[0].green = 0xcccc;
	pref->mpref[pref->mprefSize]->fore[0].blue =  0xcccc;

	pref->mpref[pref->mprefSize]->back[0].red =   0x0000;
	pref->mpref[pref->mprefSize]->back[0].green = 0x0000;
	pref->mpref[pref->mprefSize]->back[0].blue =  0x0000;

	pref->mpref[pref->mprefSize]->colors[0].red   = 0x0000;
	pref->mpref[pref->mprefSize]->colors[0].green = 0x0000;
	pref->mpref[pref->mprefSize]->colors[0].blue  = 0x0000;

	pref->mpref[pref->mprefSize]->colors[1].red   = 0xaaaa;
	pref->mpref[pref->mprefSize]->colors[1].green = 0x0000;
	pref->mpref[pref->mprefSize]->colors[1].blue  = 0x0000;

	pref->mpref[pref->mprefSize]->colors[2].red   = 0x0000;
	pref->mpref[pref->mprefSize]->colors[2].green = 0xaaaa;
	pref->mpref[pref->mprefSize]->colors[2].blue  = 0x0000;

	pref->mpref[pref->mprefSize]->colors[3].red   = 0xaaaa;
	pref->mpref[pref->mprefSize]->colors[3].green = 0x5555;
	pref->mpref[pref->mprefSize]->colors[3].blue  = 0x0000;

	pref->mpref[pref->mprefSize]->colors[4].red   = 0x6666;
	pref->mpref[pref->mprefSize]->colors[4].green = 0x6666;
	pref->mpref[pref->mprefSize]->colors[4].blue  = 0xffff;

	pref->mpref[pref->mprefSize]->colors[5].red   = 0xaaaa;
	pref->mpref[pref->mprefSize]->colors[5].green = 0x0000;
	pref->mpref[pref->mprefSize]->colors[5].blue  = 0xaaaa;

	pref->mpref[pref->mprefSize]->colors[6].red   = 0x0000;
	pref->mpref[pref->mprefSize]->colors[6].green = 0xaaaa;
	pref->mpref[pref->mprefSize]->colors[6].blue  = 0xaaaa;

	pref->mpref[pref->mprefSize]->colors[7].red   = 0x0000;
	pref->mpref[pref->mprefSize]->colors[7].green = 0xcccc;
	pref->mpref[pref->mprefSize]->colors[7].blue  = 0x0000;

	pref->mpref[pref->mprefSize]->colors[8].red   = 0x5555;
	pref->mpref[pref->mprefSize]->colors[8].green = 0x5555;
	pref->mpref[pref->mprefSize]->colors[8].blue  = 0x5555;

	pref->mpref[pref->mprefSize]->colors[9].red   = 0xffff;
	pref->mpref[pref->mprefSize]->colors[9].green = 0x5555;
	pref->mpref[pref->mprefSize]->colors[9].blue  = 0x5555;

	pref->mpref[pref->mprefSize]->colors[10].red   = 0x5555;
	pref->mpref[pref->mprefSize]->colors[10].green = 0xffff;
	pref->mpref[pref->mprefSize]->colors[10].blue  = 0x5555;

	pref->mpref[pref->mprefSize]->colors[11].red   = 0xffff;
	pref->mpref[pref->mprefSize]->colors[11].green = 0xffff;
	pref->mpref[pref->mprefSize]->colors[11].blue  = 0x5555;

	pref->mpref[pref->mprefSize]->colors[12].red   = 0x5555;
	pref->mpref[pref->mprefSize]->colors[12].green = 0x5555;
	pref->mpref[pref->mprefSize]->colors[12].blue  = 0xffff;

	pref->mpref[pref->mprefSize]->colors[13].red   = 0xffff;
	pref->mpref[pref->mprefSize]->colors[13].green = 0x5555;
	pref->mpref[pref->mprefSize]->colors[13].blue  = 0xffff;

	pref->mpref[pref->mprefSize]->colors[14].red   = 0x5555;
	pref->mpref[pref->mprefSize]->colors[14].green = 0xffff;
	pref->mpref[pref->mprefSize]->colors[14].blue  = 0xffff;

	pref->mpref[pref->mprefSize]->colors[15].red   = 0xffff;
	pref->mpref[pref->mprefSize]->colors[15].green = 0xffff;
	pref->mpref[pref->mprefSize]->colors[15].blue  = 0xffff;

	return pref->mpref[pref->mprefSize];
}

int gtkTermPref_save(gtkTermPref* pref, GString *path)
{
	FILE* gtkTermRC_FP;
	int i;

	gtkTermRC_FP = fopen (path->str, "w");
	if(!gtkTermRC_FP)
	{
		g_error("GtkTerm could not write the pref: %s", path->str);
		return FALSE;
	}
	fprintf(gtkTermRC_FP, "##\n# 1=RIGHT, 2=TOP, 3=BOTTOM, 4=LEFT\n##\n");
	fprintf(gtkTermRC_FP, "notebookPanel=%d\n\n", pref->notebookPanel);
	fprintf(gtkTermRC_FP, "##\n# 0=HIDDEN, 1=LEFT, 2=RIGHT\n##\n");
	fprintf(gtkTermRC_FP, "terminalScrollbar=%d\n\n", pref->terminalScrollbar);
	fprintf(gtkTermRC_FP, "##\n# 0=NO SCROLLBAR\n##\n");
	fprintf(gtkTermRC_FP, "maxScrollbackBuffer=%d\n\n", pref->maxScrollbackBuffer);
	fprintf(gtkTermRC_FP, "##\n# 0=FALSE, 1=TRUE\n##\n");
	fprintf(gtkTermRC_FP, "quitPRGonExitLastTerminal=%d\n\n", pref->quitPRGonExitLastTerminal);
	fprintf(gtkTermRC_FP, "##\n# 0=FALSE, 1=TRUE\n##\n");
	fprintf(gtkTermRC_FP, "showTabsByOneTerminal=%d\n\n", pref->showTabsByOneTerminal);
	fprintf(gtkTermRC_FP, "##\n#A Font name: use gnome-font-properties to look at available fonts\n##\n");
	fprintf(gtkTermRC_FP, "terminalFont=%s\n\n", pref->terminalFont);
	fprintf(gtkTermRC_FP, "##\n# Separator for marking with mouse\n##\n");
	fprintf(gtkTermRC_FP, "worldClass=%s\n\n", pref->worldClass);
	fprintf(gtkTermRC_FP, "##\n# 0=FALSE, 1=TRUE\n##\n");
	fprintf(gtkTermRC_FP, "scrollOnKeyStroke=%d\n\n", pref->scrollOnKeyStroke);
	fprintf(gtkTermRC_FP, "##\n# 0=FALSE, 1=TRUE\n##\n");
	fprintf(gtkTermRC_FP, "scrollOnOutput=%d\n\n", pref->scrollOnOutput);
	fprintf(gtkTermRC_FP, "##\n# 0=FALSE, 1=TRUE\n##\n");
	fprintf(gtkTermRC_FP, "blink=%d\n\n", pref->blink);
	fprintf(gtkTermRC_FP, "##\n# 0=FALSE, 1=TRUE\n##\n");
	fprintf(gtkTermRC_FP, "beep=%d\n\n", pref->beep);
	fprintf(gtkTermRC_FP, "##\n# x y Default Terminal Size in pix\n##\n");
	fprintf(gtkTermRC_FP, "terminalSize=%d %d\n\n", pref->termX, pref->termY);
	fprintf(gtkTermRC_FP, "##\n# NEW COLOR Definition\n##\n");
	fprintf(gtkTermRC_FP, "red=0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x\n",   pref->back[0].red, pref->fore[0].red, pref->colors[0].red, pref->colors[1].red, pref->colors[2].red, pref->colors[3].red, pref->colors[4].red, pref->colors[5].red, pref->colors[6].red, pref->colors[7].red, pref->colors[8].red, pref->colors[9].red, pref->colors[10].red, pref->colors[11].red, pref->colors[12].red, pref->colors[13].red, pref->colors[14].red, pref->colors[15].red);
	fprintf(gtkTermRC_FP, "grn=0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x\n",   pref->back[0].green, pref->fore[0].green, pref->colors[0].green, pref->colors[1].green, pref->colors[2].green, pref->colors[3].green, pref->colors[4].green, pref->colors[5].green, pref->colors[6].green, pref->colors[7].green, pref->colors[8].green, pref->colors[9].green, pref->colors[10].green, pref->colors[11].green, pref->colors[12].green, pref->colors[13].green, pref->colors[14].green, pref->colors[15].green);
	fprintf(gtkTermRC_FP, "blu=0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x\n\n", pref->back[0].blue, pref->fore[0].blue, pref->colors[0].blue, pref->colors[1].blue, pref->colors[2].blue, pref->colors[3].blue, pref->colors[4].blue, pref->colors[5].blue, pref->colors[6].blue, pref->colors[7].blue, pref->colors[8].blue, pref->colors[9].blue, pref->colors[10].blue, pref->colors[11].blue, pref->colors[12].blue, pref->colors[13].blue, pref->colors[14].blue, pref->colors[15].blue);
	fprintf(gtkTermRC_FP, "##\n# COLOR Definition\n##\n");
	fprintf(gtkTermRC_FP, "\n##\n# First section\n##\n");

	if(pref->mprefSize > 0 && pref->mpref != NULL)
	{
		for(i=1; i<=pref->mprefSize; i++)
		{
			fprintf(gtkTermRC_FP, "[section]\n");
			fprintf(gtkTermRC_FP, "match=%s\n", pref->mpref[i]->match);
			fprintf(gtkTermRC_FP, "worldClass=%s\n", pref->mpref[i]->worldClass);
			fprintf(gtkTermRC_FP, "scrollOnKeyStroke=%d\n", pref->mpref[i]->scrollOnKeyStroke);
			fprintf(gtkTermRC_FP, "scrollOnOutput=%d\n", pref->mpref[i]->scrollOnOutput);
			fprintf(gtkTermRC_FP, "beep=%d\n", pref->mpref[i]->beep);
			fprintf(gtkTermRC_FP, "blink=%d\n", pref->mpref[i]->blink);
			fprintf(gtkTermRC_FP, "red=0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x\n",   pref->mpref[i]->back[0].red, pref->mpref[i]->fore[0].red, pref->mpref[i]->colors[0].red,   pref->mpref[i]->colors[1].red, pref->mpref[i]->colors[2].red, pref->mpref[i]->colors[3].red, pref->mpref[i]->colors[4].red, pref->mpref[i]->colors[5].red, pref->mpref[i]->colors[6].red, pref->mpref[i]->colors[7].red, pref->mpref[i]->colors[8].red, pref->mpref[i]->colors[9].red, pref->mpref[i]->colors[10].red, pref->mpref[i]->colors[11].red, pref->mpref[i]->colors[12].red, pref->mpref[i]->colors[13].red, pref->mpref[i]->colors[14].red, pref->mpref[i]->colors[15].red);
			fprintf(gtkTermRC_FP, "grn=0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x\n",   pref->mpref[i]->back[0].green, pref->mpref[i]->fore[0].green, pref->mpref[i]->colors[0].green, pref->mpref[i]->colors[1].green, pref->mpref[i]->colors[2].green, pref->mpref[i]->colors[3].green, pref->mpref[i]->colors[4].green, pref->mpref[i]->colors[5].green, pref->mpref[i]->colors[6].green, pref->mpref[i]->colors[7].green, pref->mpref[i]->colors[8].green, pref->mpref[i]->colors[9].green, pref->mpref[i]->colors[10].green, pref->mpref[i]->colors[11].green, pref->mpref[i]->colors[12].green, pref->mpref[i]->colors[13].green, pref->mpref[i]->colors[14].green, pref->mpref[i]->colors[15].green);
			fprintf(gtkTermRC_FP, "blu=0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x 0x%04x\n\n", pref->mpref[i]->back[0].blue, pref->mpref[i]->fore[0].blue, pref->mpref[i]->colors[0].blue, pref->mpref[i]->colors[1].blue, pref->mpref[i]->colors[2].blue, pref->mpref[i]->colors[3].blue, pref->mpref[i]->colors[4].blue, pref->mpref[i]->colors[5].blue, pref->mpref[i]->colors[6].blue, pref->mpref[i]->colors[7].blue, pref->mpref[i]->colors[8].blue, pref->mpref[i]->colors[9].blue, pref->mpref[i]->colors[10].blue, pref->mpref[i]->colors[11].blue, pref->mpref[i]->colors[12].blue, pref->mpref[i]->colors[13].blue, pref->mpref[i]->colors[14].blue, pref->mpref[i]->colors[15].blue);
		}
	}

	fclose (gtkTermRC_FP);

	return TRUE;
}

int gtkTermPref_free(gtkTermPref* pref)
{
	int i;

	for(i=0; i<pref->mprefSize; i++)
	{
		free(pref->mpref[i]);
	}

	free(pref->mpref);
	free(pref);
	pref = NULL;

	return TRUE;
}

gtkTermPref* gtkTermPref_get (void)
{
	FILE* gtkTermRC;
	GString* gtktermrc_file;
	gtkTermPref* pref;
	char tmp[MAX_LINE_LENGTH];
	char* worldClass;
	int i=0;
	int* blink;
	int* beep;
	int* scrollOnOutput;
	int* scrollOnKeyStroke;
	GdkColor *colors;
	GdkColor *fore;
	GdkColor *back;
	char *home;

	pref = gtkTermPref_init();

	gtktermrc_file = g_string_new("");
	home = getenv("HOME");
	if (home == NULL) {
		home = "";
	}
	g_string_sprintf(gtktermrc_file, "%s/.gtkterm2rc", home);

	gtkTermRC = fopen (gtktermrc_file->str, "r");
	if(!gtkTermRC)
	{
		g_warning("cannot open gtkterm2rc file \"%s\": %s\n",
			gtktermrc_file->str, g_strerror(errno));
		if(gtkTermPref_save(pref, gtktermrc_file) == FALSE)
		{
			gtkTermPref_free(pref);
			g_string_free(gtktermrc_file, TRUE);
			exit(EXIT_FAILURE);
		}
		else if(!(gtkTermRC = fopen (gtktermrc_file->str, "r")))
		{
			g_warning("cannot open gtkterm2rc file \"%s\": %s\n",
				gtktermrc_file->str, g_strerror(errno));
			exit(EXIT_FAILURE);
		}
	}

	blink = &pref->blink;
	beep  = &pref->beep;
	scrollOnOutput = &pref->scrollOnOutput;
	scrollOnKeyStroke = &pref->scrollOnKeyStroke;
	worldClass = pref->worldClass;

	colors = (GdkColor *) &pref->colors;
	back   = (GdkColor *) &pref->back;
	fore   = (GdkColor *) &pref->fore;

	while (!feof (gtkTermRC))
	{
		fgets(tmp, MAX_LINE_LENGTH - 1, gtkTermRC);
		if(tmp[0] != '#' && tmp[0] != '\n' && tmp[0] != ' ')
		{
			if(strstr(tmp, "[section]"))
			{
				i++;
				if(i > pref->mprefSize)
				{
					gtkTermMPref_init(pref);
				}
				blink = &pref->mpref[i]->blink;
				beep  = &pref->mpref[i]->beep;
				scrollOnOutput = &pref->mpref[i]->scrollOnOutput;
				scrollOnKeyStroke = &pref->mpref[i]->scrollOnKeyStroke;
				worldClass = pref->mpref[i]->worldClass;

				colors = (GdkColor *) &pref->mpref[i]->colors;
				back   = (GdkColor *) &pref->mpref[i]->back;
				fore   = (GdkColor *) &pref->mpref[i]->fore;
			}
			if(strstr(tmp, "notebookPanel"))
			{
				sscanf(tmp + 14, "%d", (int *) &pref->notebookPanel);
			}
			if(strstr(tmp, "terminalScrollbar"))
			{
				sscanf(tmp + 18, "%d", (int *) &pref->terminalScrollbar);
			}
			if(strstr(tmp, "maxScrollbackBuffer"))
			{
				sscanf(tmp + 20, "%d", (int *) &pref->maxScrollbackBuffer);
			}
			if(strstr(tmp, "quitPRGonExitLastTerminal"))
			{
				sscanf(tmp + 26, "%d", (int *) &pref->quitPRGonExitLastTerminal);
			}
			if(strstr(tmp, "showTabsByOneTerminal"))
			{
				sscanf(tmp + 22, "%d", (int *) &pref->showTabsByOneTerminal);
			}
			if(strstr(tmp, "terminalFont"))
			{
				sscanf(tmp + 13, "%[0123456789 abcdefghijklmnopqrstuwvxyzABCDEFGHIJKLMNOPQRSTUWVXYZ]", pref->terminalFont);
			}
			if(strstr(tmp, "worldClass"))
			{
				sscanf(tmp + 11, "%s", worldClass);
			}
			if(strstr(tmp, "scrollOnOutput"))
			{
				sscanf(tmp + 15, "%d", scrollOnOutput);
			}
			if(strstr(tmp, "scrollOnKeyStroke"))
			{
				sscanf(tmp + 18, "%d", scrollOnKeyStroke);
			}
			if(strstr(tmp, "blink"))
			{
				sscanf(tmp +  6, "%d", blink);
			}
			if(strstr(tmp, "beep"))
			{
				sscanf(tmp +  5, "%d", beep);
			}
			if(strstr(tmp, "terminalSize"))
			{
				sscanf(tmp + 13, "%d %d", (int *) &pref->termX, (int *) &pref->termY);
			}
			if(strstr(tmp, "red"))
			{
				sscanf(tmp + 4, "%hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi", &back[0].red, &fore[0].red, &colors[0].red, &colors[1].red, &colors[2].red, &colors[3].red, &colors[4].red, &colors[5].red, &colors[6].red, &colors[7].red, &colors[8].red, &colors[9].red, &colors[10].red, &colors[11].red, &colors[12].red, &colors[13].red, &colors[14].red, &colors[15].red);
			}
			if(strstr(tmp, "grn"))
			{
				sscanf(tmp + 4, "%hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi", &back[0].green, &fore[0].green, &colors[0].green, &colors[1].green, &colors[2].green, &colors[3].green, &colors[4].green, &colors[5].green, &colors[6].green, &colors[7].green, &colors[8].green, &colors[9].green, &colors[10].green, &colors[11].green, &colors[12].green, &colors[13].green, &colors[14].green, &colors[15].green);
			}
			if(strstr(tmp, "blu"))
			{
				sscanf(tmp + 4, "%hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi %hi", &back[0].blue, &fore[0].blue, &colors[0].blue, &colors[1].blue, &colors[2].blue, &colors[3].blue, &colors[4].blue, &colors[5].blue, &colors[6].blue, &colors[7].blue, &colors[8].blue, &colors[9].blue, &colors[10].blue, &colors[11].blue, &colors[12].blue, &colors[13].blue, &colors[14].blue, &colors[15].blue);
			}
			if(strstr(tmp, "match"))
			{
				sscanf(tmp + 6, "%s", pref->mpref[i]->match);
			}
		}
	}

	fclose(gtkTermRC);
	if(gtkTermPref_save(pref, gtktermrc_file) == FALSE)
	{
		gtkTermPref_free(pref);
		g_string_free(gtktermrc_file, TRUE);
		exit(EXIT_FAILURE);
	}
	g_string_free(gtktermrc_file, TRUE);

	return pref;
}

int gtkTermFindMPref(char *matchstr, gtkTermPref *pref)
{
	int i;

	for(i=1; i<=pref->mprefSize; i++)
	{
		if(strstr(matchstr, pref->mpref[i]->match))
		{
			return i;
		}
	}
	return -1;
}

void gtkTermSetMPref(GtkWidget *term, gtkTermPref *pref, int i)
{
	gpointer termPref;

	if((termPref = gtk_object_get_data(GTK_OBJECT(term), "mpref")) == NULL)
	{
		gtk_object_set_data(GTK_OBJECT(term), "mpref", GINT_TO_POINTER(-1));
		termPref = GINT_TO_POINTER(-1);
	}

	if(GPOINTER_TO_INT(termPref) != i)
	{
		if(i == -1)
		{
			vte_terminal_set_audible_bell(VTE_TERMINAL(term), pref->beep);
			vte_terminal_set_cursor_blinks(VTE_TERMINAL(term), pref->blink);
			vte_terminal_set_colors(VTE_TERMINAL(term), &pref->fore[0], &pref->back[0], pref->colors, 16);
			vte_terminal_set_scroll_on_output(VTE_TERMINAL(term), pref->scrollOnOutput);
			vte_terminal_set_scroll_on_keystroke(VTE_TERMINAL(term), pref->scrollOnKeyStroke);
			vte_terminal_set_word_chars (VTE_TERMINAL(term), pref->worldClass);
		}
		else
		{
			vte_terminal_set_audible_bell(VTE_TERMINAL(term), pref->mpref[i]->beep);
			vte_terminal_set_cursor_blinks(VTE_TERMINAL(term), pref->mpref[i]->blink);
			vte_terminal_set_colors(VTE_TERMINAL(term), &pref->mpref[i]->fore[0], &pref->mpref[i]->back[0], pref->mpref[i]->colors, 16);
			vte_terminal_set_scroll_on_output(VTE_TERMINAL(term), pref->mpref[i]->scrollOnOutput);
			vte_terminal_set_scroll_on_keystroke(VTE_TERMINAL(term), pref->mpref[i]->scrollOnKeyStroke);
			vte_terminal_set_word_chars (VTE_TERMINAL(term), pref->mpref[i]->worldClass);
		}
		gtk_object_set_data(GTK_OBJECT(term), "mpref", GINT_TO_POINTER(i));
	}
}
